caikit_nlp/modules/reranker/rerank.py

```diff
@@ -0,0 +1,198 @@
+import os
+
+# Third Party
+from sentence_transformers import SentenceTransformer
+from sentence_transformers.util import dot_score, normalize_embeddings, semantic_search
+
+# First Party
+from caikit.core import ModuleBase, ModuleConfig, ModuleSaver, module
+from caikit.core.data_model.json_dict import JsonDict
+from caikit.core.exceptions import error_handler
+import alog
+
+# Local
+from .rerank_task import RerankTask
+from caikit_nlp.data_model.reranker import (
+    RerankPrediction,
+    RerankQueryResult,
+    RerankScore,
+)
+
+logger = alog.use_channel("<EMBD_BLK>")
+error = error_handler.get(logger)
+
+
+@module(
+    "00110203-0405-0607-0809-0a0b02dd0e0f",
+    "RerankerModule",
+    "0.0.1",
+    RerankTask,
+)
+class Rerank(ModuleBase):
+
+    _ARTIFACTS_PATH_KEY = "artifacts_path"
+    _ARTIFACTS_PATH_DEFAULT = "artifacts"
+    _HF_HUB_KEY = "hf_model"
+
+    def __init__(
+        self,
+        model: SentenceTransformer,
+    ):
+        """Initialize
+        This function gets called by `.load` and `.train` function
+        which initializes this module.
+        """
+        super().__init__()
+        self.model = model
+
+    @classmethod
+    def load(cls, model_path: str) -> "Rerank":
+        """Load a model
+
+        Args:
+            model_path: str
+                Path to the config dir of the model to be loaded.
+
+        Returns:
+            Rerank
+                Instance of this class built from the model.
+        """
+
+        config = ModuleConfig.load(model_path)
+
+        artifacts_path = config.get(cls._ARTIFACTS_PATH_KEY)
+        if artifacts_path:
+            model_name_or_path = os.path.abspath(
+                os.path.join(model_path, artifacts_path)
+            )
+            error.dir_check("<NLP01027299E>", model_name_or_path)
+        else:
+            # If no artifacts_path, look for hf_model Hugging Face model by name (or path)
+            model_name_or_path = config.get(cls._HF_HUB_KEY)
+            error.value_check(
+                "<NLP22444208E>",
+                model_name_or_path,
+                ValueError(
+                    f"Model config missing '{cls._ARTIFACTS_PATH_KEY}' or '{cls._HF_HUB_KEY}'"
+                ),
+            )
+
+        return cls.bootstrap(model_name_or_path=model_name_or_path)
+
+    def run(
+        self,
+        queries: List[str],
+        documents: List[JsonDict],
+        top_n: Optional[int] = None,
+    ) -> RerankPrediction:
+        """Run inference on model.
+        Args:
+            queries: List[str]
+            documents:  List[JsonDict]
+            top_n:  Optional[int]
+        Returns:
+            RerankPrediction
+        """
+
+        error.type_check(
+            "<NLP09038249E>",
+            list,
+            queries=queries,
+            documents=documents,
+        )
+
+        if len(queries) < 1 or len(documents) < 1:
+            return RerankPrediction([])
+
+        if top_n is None or top_n < 1:
```

Comment:
this default behavior is undocumented. Can we please add this in docstrings?

Reply:
done